src/main.ts

```diff
@@ -237,7 +260,43 @@ export class Main {
     ipcMain.on('setup-update-interval', () => { this.setupUpdateInterval() });
     ipcMain.on('clear-update-interval', () => { this.clearUpdateInterval() });
 
-    this.omnitruckUpdateChecker.on('start-update-check', () => {
+    // @pranay
+    // @ts-ignore
+    ipcMain.on('select-dirs', async (event, arg) => {
+      console.log('directories selected', event)
+      console.log('directories selected', arg)
+      const { canceled, filePaths } = await dialog.showOpenDialog(this.backgroundWindow, {
+        properties: ['openDirectory']
+      })
+      console.log('directories selected', filePaths[0])
+        if (canceled) {
+          return ""
+        } else {
+            let os = require('os'),
+            fs = require('fs'),
+            path = require('path');
+            let reposFile = path.join(os.homedir(), '.chef/repository.json')
+            console.log("reposFile", reposFile)
+            if(!fs.existsSync(reposFile)){
+              console.log("reposFile not found")
+              helpers.createChefReposJson();
+            }
+          if (!this.checkForDuplicate(filePaths[0])){
+            console.log("returning new cookbook from here")
+            // append this file path to json
+            helpers.writeRepoPath( filePaths[0], "local")
+            // send back cookname and append it to file or refresh whole page
+            event.reply("select-dirs-response", filePaths[0])
+            return filePaths[0]
+          } else{
+            console.log("folder is already present")
```

Comment:
can we make the return arg more structured so we can easily discern between success and failure (in this case duplicate folder)?

Reply:
Yes, returned something like  `event.reply("select-dirs-response", filePaths[0]);` which is received at renderer and is shown at alert.